osu.Game/Overlays/BeatmapSet/ExplicitBeatmapPill.cs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+using osuTK;
+using osuTK.Graphics;
+
+namespace osu.Game.Overlays.BeatmapSet
+{
+    public class ExplicitBeatmapPill : CompositeDrawable
+    {
+        public ExplicitBeatmapPill()
+        {
+            AutoSizeAxes = Axes.Both;
+        }
+
+        [BackgroundDependencyLoader(true)]
+        private void load(OsuColour colours, OverlayColourProvider colourProvider)
+        {
+            InternalChild = new CircularContainer
+            {
+                Masking = true,
+                AutoSizeAxes = Axes.Both,
+                Children = new Drawable[]
+                {
+                    new Box
+                    {
+                        RelativeSizeAxes = Axes.Both,
+                        Colour = colourProvider?.Background5 ?? colours.Gray2,
+                    },
+                    new OsuSpriteText
+                    {
+                        Margin = new MarginPadding { Horizontal = 10f, Vertical = 2f },
+                        Text = "EXPLICIT",
+                        Font = OsuFont.GetFont(size: 10, weight: FontWeight.Bold),
+                        // todo: this is --hsl-orange-2 from the new palette in https://github.com/ppy/osu-web/blob/8ceb46f/resources/assets/less/colors.less#L128-L151,
```

Comment:
Implement in `OverlayColourProvider` as 1/2/3/4

![image](https://user-images.githubusercontent.com/191335/104460103-01c37c80-55f1-11eb-9050-1ce5ee630470.png)

Reply:
Wasn't sure about adding it in `OverlayColourProvider` at first, but seems like it shouldn't have been dedicated to overlays only.

As for accessing it, you fine with `new OverlayColourProvider(OverlayColourScheme.Orange).Colour2`? thinking about making `static readonly` fields for each theme instead.

Reply:
If we're not in an orange scheme then yeah, that's the best way for now.